var/spack/repos/builtin/packages/metis/package.py

```diff
@@ -95,6 +95,13 @@ def install(self, spec, prefix):
         if any('+{0}'.format(v) in spec for v in ['gdb', 'int64', 'real64']):
             raise InstallError('METIS@:4 does not support the following '
                                'variants: gdb, int64, real64.')
+        # This build system only supports 'Release' or 'Debug' build_types.
+        if spec.variants['build_type'].value != 'Release' and \
+                spec.variants['build_type'].value != 'Debug':
+            raise InstallError('METIS@:4 only supports build_type=Release or '
```

Comment:
sorry to be a PITA, but please use `conflicts('@:4.999', when='buid_type=...')` and alike as this will throw the error at concretization, whereas your code raises error during installation.

The code with `raise InstallError` above probably was written before `conflicts()` statement was available.